### PR TITLE
XIMEA adapter update

### PR DESCRIPTION
**Pull request by mzajko**
- replaced m3api/m3apiX64 binaries with latest xiapi32/xiapi64 to
support latest camera models
- added new parameter definitions to videoio_c.h and
SetProperty/GetProperty calls
- added acquisition resetting for specific parmateters in SetProperty
call
- added new error return value evaluators

### CODE REVIEW

**Comment by mzajko** [comment]
Hi All,

so this last PR seems to have failed only on the whitespace check. Please advise what should be
fixed in the sources to pass it. 
@alalek - Thank you very much fro your help on this so far!

**Comment by mzajko** [comment]
Hi All,

I managed to fix the whitespace issues, but now it is failing on OpenCL.
Please advise what can be done to fix these issues. Thanks.

File: modules/videoio/include/opencv2/videoio/videoio_c.h [hunk 1]
```diff
@@ -240,27 +240,147 @@ enum
     CV_CAP_PROP_PVAPI_PIXELFORMAT           = 306, // Pixel format
 
     // Properties of cameras available through XIMEA SDK interface
-    CV_CAP_PROP_XI_DOWNSAMPLING  = 400,      // Change image resolution by binning or skipping.
```
Comment (vpisarev) [inline modules/videoio/include/opencv2/videoio/videoio_c.h]: master now represents updates to the stable 3.0 release, where we want to preserve binary compatibility. Changing numerical values of the enumeration constants obviously breaks binary compatibility. Could you please retain the original list as-is and simply add the new values in the end?

**Comment by mzajko** [comment]
Hi Vadim,

thank you for your comments, it should be fixed now.

**Comment by vpisarev** [comment]
@mzajko, thanks! now it's much better. As I see, the CV_CAP_PROP_XI_DOWNSAMPLING constant is still not the same as before.

**Comment by mzajko** [comment]
Hi Vadim,

sorry for that, slipped my eye... It should be fixed now.

**Comment by mzajko** [comment]
Hi All,

can someone please explain what happened in the last buildbot run for this PR?
Thanks a lot.

**Comment by vpisarev** [comment]
:+1:

**Comment by mzajko** [comment]
Hi, when do you plan to merge the updated sources?

**Pull request merged**